test/unittest/dds/participant/ParticipantTests.cpp

```diff
@@ -49,10 +49,10 @@ TEST(ParticipantTests, DomainParticipantListenerMethods)
 
     ASSERT_EQ(participant->get_listener(), nullptr);
 
-    DomainParticipantListener* listener;
-    participant->set_listener(listener);
+    DomainParticipantListener listener = DomainParticipantListener();
```

Comment:
```suggestion
    DomainParticipantListener listener;
```
No need to construct-and-assign.